hawkular-wildfly-agent-installer/src/main/java/org/hawkular/wildfly/agent/installer/AgentInstaller.java

```diff
@@ -349,8 +352,21 @@ private static String createSecurityRealm(String keystoreFile, String keystorePa
      */
     private static XmlEdit createStorageAdapter(boolean withHttps, InstallerConfiguration installerConfig) {
         String select = "/server/profile/*[namespace-uri()='urn:org.hawkular.agent:agent:1.0']/";
-        StringBuilder xml = new StringBuilder("<storage-adapter")
-                .append(" type=\"HAWKULAR\"");
+        StringBuilder xml = new StringBuilder("<storage-adapter");
+
+        String tenantId = installerConfig.getTenantId();
+        if (installerConfig.isMetricsOnlyMode()) {
+            xml.append(" type=\"METRICS\"");
+            if (tenantId == null || tenantId.isEmpty()) {
+                throw new IllegalArgumentException("You must specify tenant-id when in metrics-only mode");
+            }
+        } else {
+            xml.append(" type=\"HAWKULAR\"");
+        }
+
+        if (tenantId != null && !tenantId.isEmpty()) {
+            xml.append(" tenantId=\"" + tenantId + "\"");
```

Comment:
yes, this allows a tenant ID to be specified for HAWKULAR mode. What this allows is when the agent starts up, IFF the tenant ID matches that of the agent's credentials, the agent starts up fine. If the tenantId configured here does NOT match the user's credentials, the agent will log an error and not start. So it works as a sort of "check" - kind of like allowing the person installing the agent to say, "I expect the agent to be associated with tenant ID "foo" - if the credentials given to the agent do not give it tenant Id "foo" the agent should not start."

We could throw an exception here if we don't like that "feature" - but, I didn't see any harm in allowing this check-tenant feature so I left it here.